### PR TITLE
test/pod: add script for running each test pass in a separate namespace

### PR DESCRIPTION
**Pull request by hasbro17**
[skip ci]

The `run-test-passes` script runs each test pass in a separate pod and namespace. It saves the logs for each pass to a file. 
After waiting for all passes to complete it prints out the logs for all failed passes.

/cc @hongchaodeng @fanminshi

### CODE REVIEW

File: test/pod/run-test-pod [hunk 1]
```diff
@@ -76,12 +75,3 @@ echo "test-pod logs"
 echo "=============="
 kubectl -n ${TEST_NAMESPACE} logs -f ${POD_NAME}
 echo "=============="
-
-# Check for pod success or failure
```
Comment (hongchaodeng) [inline test/pod/run-test-pod]: Why delete them?

File: test/pod/run-test-passes [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+for p in $PASSES; do
+    echo "${p} pass"
+
+    # Create the pass namespace
+    PASS_NAMESPACE=${p}-${ROOT_NAMESPACE}
+    kubectl create ns ${PASS_NAMESPACE}
+
+    # Copy the aws secret into pass namespace
+    kubectl -n ${ROOT_NAMESPACE} get secrets ${TEST_AWS_SECRET} -o yaml | sed "s|${ROOT_NAMESPACE}|${PASS_NAMESPACE}|g" | kubectl create -f -
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: sed "s|namespace: xxx | namespace: xxx|g"

File: test/pod/run-test-passes [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    # Run the run-test-pod script for this pass in the pass namespace and save the logs
+    POD_NAME=${p}-test-pod
+	echo "Running test-pod (${POD_NAME}) in namespace (${PASS_NAMESPACE})"
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: indent

File: test/pod/run-test-passes [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+# Cleanup for the per pass namespace
+function cleanup_namespaces {
+    for p in ${PASSES}; do
+        PASS_NAMESPACE=${p}-${ROOT_NAMESPACE}
+        kubectl delete ns ${PASS_NAMESPACE} || true
+    done
+}
+trap cleanup_namespaces EXIT
+
+for p in $PASSES; do
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: Better rename `PASSES` inside the script?

File: test/pod/run-test-passes [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    kubectl create ns ${PASS_NAMESPACE}
+
+    # Copy the aws secret into pass namespace
+    kubectl -n ${ROOT_NAMESPACE} get secrets ${TEST_AWS_SECRET} -o yaml | sed "s|${ROOT_NAMESPACE}|${PASS_NAMESPACE}|g" | kubectl create -f -
+
+    # Run the run-test-pod script for this pass in the pass namespace and save the logs
+    POD_NAME=${p}-test-pod
+	echo "Running test-pod (${POD_NAME}) in namespace (${PASS_NAMESPACE})"
+	PASSES=${p} \
+    POD_NAME=${POD_NAME} \
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: indent:
```
POD_NAME=xxx \
    TEST_NAMESPACE=xxx \
    ...
```

File: test/pod/run-test-passes [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    TEST_NAMESPACE=${PASS_NAMESPACE} \
+    TEST_IMAGE=${TEST_IMAGE} \
+    TEST_S3_BUCKET=${TEST_S3_BUCKET} \
+    TEST_AWS_SECRET=${TEST_AWS_SECRET} \
+    OPERATOR_IMAGE=${OPERATOR_IMAGE} \
+    test/pod/run-test-pod > ${LOG_DIR}/${p}_pass.log 2>&1 &
+    echo ""
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: what's this for?
Seems unneeded. Remove it?

File: test/pod/run-test-passes [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+done
+
+echo "Waiting for all passes to complete..."
+wait
```
Comment (hongchaodeng) [inline test/pod/run-test-passes]: It seems a bad idea to do concurrency control in bash script.
We should use a go program.
But again, I think this scope is bigger than needed. Just do `e2e` initially without caring about concurrency.

**Comment by hongchaodeng** [comment]
@hasbro17 
Let's not deal with two concurrent jobs initially, but only handle `e2e` alone.
We should discuss concurrent jobs and define what to be done in https://github.com/coreos/etcd-operator/issues/1640

**Comment by hasbro17** [comment]
@hongchaodeng Sure let me follow this up on the issue, describe the current status, and we can decide how to proceed from there.

**Comment by hasbro17** [comment]
Closing this until we decide on a strategy for running the phases concurrently.

**Pull request closed without merging**